java/org/contikios/cooja/contikimote/ContikiMoteType.java

```diff
@@ -228,7 +228,7 @@ private File getMoteFile(String extension) {
   }
 
   @Override
-  public LinkedHashMap<String, String> getCompilationEnvironment() {
+  public Map<String, String> getCompilationEnvironment() {
```

Comment:
The LinkedHashMap is essential in the interface to preserve ordering,  otherwise the environment tab in AbstractCompileDialog will show things in random order. (Doesn't matter for the current mote types since they already use the LinkedHashMap in their implementations, but it matters for the next RISC-V/ARM-based mote type that someone else adds.)

Reply:
Ok, now I understand the choice of LinkedHashMap. An alternative could have been to use SortedMap or let AbstractCompileDialog sort the keys but allowing the implementation to pick the order might be useful. Thanks for the explanation.